libs/server-sdk/src/data_sources/data_source_event_handler.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <boost/asio/any_io_executor.hpp>
+
+#include "data_source_status_manager.hpp"
+#include "data_source_update_sink.hpp"
+
+#include <launchdarkly/config/shared/built/service_endpoints.hpp>
+#include <launchdarkly/context.hpp>
+#include <launchdarkly/data/evaluation_result.hpp>
+#include <launchdarkly/data_sources/data_source.hpp>
+#include <launchdarkly/logging/logger.hpp>
+
+namespace launchdarkly::server_side::data_sources {
+
+/**
+ * This class handles LaunchDarkly events, parses them, and then uses
+ * a IDataSourceUpdateSink to process the parsed events.
+ *
+ * This can be used for streaming or for polling. For polling only "put" events
+ * will be used.
+ */
+class DataSourceEventHandler {
+   public:
+    /**
+     * Status indicating if the message was processed, or if there
+     * was an issue encountered.
+     */
+    enum class MessageStatus {
+        kMessageHandled,
+        kInvalidMessage,
+        kUnhandledVerb
+    };
+
+    /**
+     * Represents a flag patch JSON from the LaunchDarkly service.
+     */
+    struct FlagPatch {
```

Comment:
This may end up being one struct, but I just put them in as placeholders. If they are two types, then the handler will use the overrides, which will be nice.